hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -174,7 +174,8 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
                              String commitActionType, Map<String, List<String>> partitionToReplaceFileIds) {
     // Create a Hoodie table which encapsulated the commits and files visible
     HoodieTable table = createTable(config, hadoopConf);
-    HoodieCommitMetadata metadata = CommitUtils.buildMetadata(stats, partitionToReplaceFileIds, extraMetadata, operationType, config.getSchema(), commitActionType);
+    HoodieCommitMetadata metadata = CommitUtils.buildMetadata(stats, partitionToReplaceFileIds,
+        extraMetadata, operationType, config.getWriteSchema(), commitActionType);
```

Comment:
so switching to `config.getWriteSchema()` here, would affect hive sync? We use the schema from the commit file to sync to hive. So if the write schema is a subset of the table schema, then we can have an issue here. Did you run into any issues like that? I think we could actually write both into the commit metadata.?

Reply:
Hi @vinothchandar ,as I described above, The `writeSchema` is the same to the table schema. So there is no negative effects to the hive sync. I have run the case in our production environment and it works well.

Reply:
Below you mentioned that `writeSchema` will be not the same as inputSchema right? I think inputSchema is what will be equal to the table schema. no?